apps/avatax/src/modules/avatax/order-confirmed/avatax-order-confirmed-payload-lines-transformer.ts

```diff
@@ -1,55 +1,40 @@
 import { LineItemModel } from "avatax/lib/models/LineItemModel";
-import {
-  DeprecatedOrderConfirmedSubscriptionFragment,
-  SaleorOrder,
-  SaleorOrderLine,
-} from "../../saleor";
-import { SaleorShippingLine } from "../../saleor/shipping-line";
-import { AvataxConfig } from "../avatax-connection-schema";
+import { ISaleorOrderConfrimedOrderLine } from "../../saleor";
+import { ISaleorOrderConfirmedShippingLine } from "../../saleor/shipping-line";
 import { AvataxTaxCodeMatches } from "../tax-code/avatax-tax-code-match-repository";
-import { AvataxOrderConfirmedTaxCodeMatcher } from "./avatax-order-confirmed-tax-code-matcher";
 
 export class AvataxOrderConfirmedPayloadLinesTransformer {
-  transform(
-    order: DeprecatedOrderConfirmedSubscriptionFragment,
-    saleorOrder: SaleorOrder,
-    config: AvataxConfig,
-    matches: AvataxTaxCodeMatches,
-  ): LineItemModel[] {
-    const isDiscounted = order.discounts.length > 0;
+  constructor(
+    private args: {
+      productLines: ISaleorOrderConfrimedOrderLine[];
+      shippingLine: ISaleorOrderConfirmedShippingLine;
+      hasShipping: boolean;
+      matches: AvataxTaxCodeMatches;
+    },
```

Comment:
This is antipattern. In constructor we should have *dependencies*. Constructor is evaluated as early as possible. We can create this class in root webhook class. In IoC frameworks it will be created when app runtime starts.

These args are something that "appears" when request is evaluated.

This class doesn't really need constructor because it has no dependencies. But your method `transform` should be pure and receive all args needed.

Why class then? You can inject dependencies later if needed. Or some static settings.
Eg. imaging we want translation. We can inject "ITranslator" to constructor and use it to translate fields.